Sources/MapboxNavigation/PreviewViewController.swift

```diff
@@ -0,0 +1,633 @@
+            speedLimitView.topAnchor.constraint(equalTo: navigationView.topBannerContainerView.bottomAnchor,
+                                                constant: 10),
+            speedLimitView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor,
+                                                    constant: 10),
+            speedLimitView.widthAnchor.constraint(equalToConstant: 50),
+            speedLimitView.heightAnchor.constraint(equalToConstant: 50)
+        ]
+        
+        NSLayoutConstraint.activate(speedLimitViewConstraints)
+        
+        self.speedLimitView = speedLimitView
+    }
+    
+    func setupWayNameView() {
+        let wayNameView: WayNameView = .forAutoLayout()
+        wayNameView.containerView.clipsToBounds = true
+        view.addSubview(wayNameView)
+        
+        let wayNameViewLayoutConstraints = [
+            wayNameView.bottomAnchor.constraint(equalTo: view.safeBottomAnchor,
+                                                constant: -60.0),
+            wayNameView.centerXAnchor.constraint(equalTo: view.safeCenterXAnchor),
+            wayNameView.widthAnchor.constraint(lessThanOrEqualTo: view.safeWidthAnchor,
+                                               multiplier: 0.95)
+        ]
+        
+        NSLayoutConstraint.activate(wayNameViewLayoutConstraints)
+        
+        self.wayNameView = wayNameView
+    }
+    
+    func setupBackButton() {
+        let backButton = BackButton(type: .system)
+        backButton.translatesAutoresizingMaskIntoConstraints = false
+        backButton.setTitle("Back", for: .normal)
+        backButton.clipsToBounds = true
+        backButton.addTarget(self, action: #selector(didPressBackButton), for: .touchUpInside)
+        
+        let backImage = UIImage(named: "back", in: .mapboxNavigation, compatibleWith: nil)!
+        backButton.setImage(backImage, for: .normal)
+        backButton.imageView?.contentMode = .scaleAspectFit
+        backButton.imageEdgeInsets = UIEdgeInsets(top: 10,
+                                                  left: 0,
+                                                  bottom: 10,
+                                                  right: 10)
+        navigationView.addSubview(backButton)
+        
+        let backButtonLayoutConstraints = [
+            backButton.widthAnchor.constraint(equalToConstant: 110),
+            backButton.heightAnchor.constraint(equalToConstant: 50),
+            backButton.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor,
+                                                constant: 10.0),
+            backButton.topAnchor.constraint(equalTo: navigationView.topBannerContainerView.bottomAnchor,
+                                            constant: 10.0)
+        ]
+        
+        NSLayoutConstraint.activate(backButtonLayoutConstraints)
+        
+        self.backButton = backButton
+    }
+    
+    @objc func didPressBackButton() {
```

Comment:
After we click the back button to the browsing mode with the waypoints all cleared. Then I find it's impossible to choose a waypoint. Because we're using the `.courseView` from the active navigation session, and the coordinate from the `passiveLocationManager` is `nil`.